runtime/lua/vim/lsp/util.lua

```diff
@@ -833,10 +833,18 @@ do
     return count
   end
   function M.buf_diagnostics_signs(bufnr, diagnostics)
-    vim.fn.sign_define('LspDiagnosticsErrorSign', {text=vim.g['LspDiagnosticsErrorSign'] or 'E', texthl='LspDiagnosticsError', linehl='', numhl=''})
-    vim.fn.sign_define('LspDiagnosticsWarningSign', {text=vim.g['LspDiagnosticsWarningSign'] or 'W', texthl='LspDiagnosticsWarning', linehl='', numhl=''})
-    vim.fn.sign_define('LspDiagnosticsInformationSign', {text=vim.g['LspDiagnosticsInformationSign'] or 'I', texthl='LspDiagnosticsInformation', linehl='', numhl=''})
-    vim.fn.sign_define('LspDiagnosticsHintSign', {text=vim.g['LspDiagnosticsHintSign'] or 'H', texthl='LspDiagnosticsHint', linehl='', numhl=''})
+    if not vim.fn.sign_getdefined('LspDiagnosticsErrorSign') then
```

Comment:
can you make a local function in this file that looks something like:

```lua
local function define_default_sign(name, properties)
  if not sign_getdefined(...) then
    ...
  end
end
```

I imagine we'll do some similar stuff later and I don't want people to forget not to constantly overwrite stuff.